src/score_predictions.py

```diff
@@ -48,53 +48,82 @@ def score_predictions(args):
 
     fold_gt, fold_pred = [], []
     for i, fold in enumerate(folds):
-        keys = ([i for i in fold.keys() if i in groundtruth.keys()])
+        keys = [i for i in fold.keys() if i in groundtruth.keys()]
         fold_pred.append([predictions[k] for k in keys])
         fold_gt.append([groundtruth[k] for k in keys])
 
-    roc_auc, pr_auc, acc, accs, report = get_metrics(
-        y_true, y_pred, fold_gt, fold_pred, n_folds)
+    roc_auc, roc_auc_std, pr_auc, pr_auc_std, acc, acc_std, report = get_metrics(
+        y_true, y_pred, fold_gt, fold_pred, n_folds
+    )
 
-    store_results(results_file, roc_auc, pr_auc, acc, accs, report, dataset)
+    store_results(
+        results_file,
+        roc_auc,
+        roc_auc_std,
+        pr_auc,
+        pr_auc_std,
+        acc,
+        acc_std,
+        report,
+        dataset,
+    )
 
-def get_metrics(y_true, y_pred, fold_gt, fold_pred, n_folds):
-    roc_auc, pr_auc = shared.compute_auc(y_true, y_pred)
-    acc = shared.compute_accuracy(y_true, y_pred)
 
+def get_metrics(y_true, y_pred, fold_gt, fold_pred, n_folds):
+    micro_acc = shared.compute_accuracy(y_true, y_pred)
     accs = []
+    roc_aucs, pr_aucs = [], []
     for i in range(n_folds):
         y_true_fold = fold_gt[i]
         y_pred_fold = fold_pred[i]
         accs.append(shared.compute_accuracy(y_true_fold, y_pred_fold))
-
-    y_true_argmax = [np.argmax(i) for i in y_true]
-    y_pred_argmax = [np.argmax(i) for i in y_pred]
-
-    report = classification_report(y_true_argmax, y_pred_argmax)
-
-    return roc_auc, pr_auc, acc, accs, report
-
-
-def store_results(output_file, roc_auc, pr_auc, acc, accs, report, dataset):
+        roc_auc_i, pr_auc_i = shared.compute_auc(y_true_fold, y_pred_fold)
+        roc_aucs.append(roc_auc_i)
+        pr_aucs.append(pr_auc_i)
+
+    macro_acc = np.mean(accs)
+    print("Macro Acc:", macro_acc)
+    print("Micro Acc:", micro_acc)
+    acc_std = np.std(accs)
+    roc_auc, pr_auc = np.mean(roc_aucs), np.mean(pr_aucs)
+    roc_auc_std, pr_auc_std = np.std(roc_aucs), np.std(pr_aucs)
+
+    y_true = np.array(y_true)
+    y_pred = np.array(y_pred)
+    if shared.type_of_groundtruth(y_true):
```

Comment:
shouldn't this be something like `shared.type_of_groundtruth(y_true) == "multilabel-indicator"`?
Maybe there's something I'm missing.